netket/vqs/mc/mc_state/state.py

```diff
@@ -338,8 +354,7 @@ def chain_length(self) -> int:
 
     @chain_length.setter
     def chain_length(self, length: int):
-        self.n_samples = length * self.sampler.n_chains * mpi.n_nodes
-        self.reset()
+        self.n_samples = length * self.sampler.n_chains
```

Comment:
Previously there was a bug that an extra `n_nodes` is multiplied. Also, we don't need to call `reset` here, because it's already called in `n_samples.setter`.